### PR TITLE
Add product change methods

### PR DESCRIPTION
**Pull request by wsmoak**
Adds product_change, delayed_product_change and cancel_delayed_product_change methods.
For https://github.com/chargify/chargify_api_ares/issues/100

### CODE REVIEW

**Comment by wsmoak** [comment]
I can make the code work, but method names and signatures in Ruby are still a challenge!

I eventually settled on:  product_change(handle), delayed_product_change(handle) and cancel_delayed_product_change.  If you want it to be subscription.change_product or something else instead, let me know.

Also, I can't figure out how to write a test for this.  Any time you call 'save' it strips out the product, and that's where the change actually happens, in subscription.product.handle.  For a delayed product change, I'm not sure if there is anything visible in the subscription object that indicates a product change is scheduled -- that might need to be added.

I have tested it with this code against a real Chargify site in test mode:
https://github.com/wsmoak/chargify/blob/master/ruby/product-change.rb

File: lib/chargify_api_ares/resources/subscription.rb [hunk 1]
```diff
@@ -138,6 +138,22 @@ def delayed_cancel(cancel_at_end_of_period = true)
       save
     end
 
+    def product_change(handle)
+      self.product_handle = handle
+      save
+    end
+
+    def delayed_product_change(handle)
```
Comment (jeremywrowe) [inline lib/chargify_api_ares/resources/subscription.rb]: Would you mind combining the `product_change` and `delayed_product_change` methods into just `product_change` with a default of `delayed = false`.

EG

```ruby
def product_change(handle, delayed = false)
  self.product_handle = handle
  self.product_change_delayed = delayed
end
```

**Comment by wsmoak** [comment]
I've made the requested changes in the code and examples.

Test code verified as working against my Chargify site (in test mode):
https://github.com/wsmoak/chargify/blob/master/ruby/product-change.rb

**Comment by jeremywrowe** [comment]
Perfect thanks @wsmoak

**Pull request merged**